### ISSUE TITLE
Excluded files still available in peek definition

### ISSUE THREAD

**Issue by zeleniy** [DEVSENSE/phptools-docs#405]
Please, see bug description here: https://github.com/microsoft/vscode/issues/194299

**Comment by jakubmisek** [comment]
Thank you for reporting the issue.

Although, as said on https://github.com/microsoft/vscode/issues/194299#issuecomment-1737721116 , the `"search.exclude"` setting is not meant for the _Peek to Definition_ functionality (or at least it's not specified, how the extension should behave)

Our extension respects `"files.exclude"` setting - the files there won't be indexed and/or navigated to.

**Comment by zeleniy** [comment]
It seems excluded files still available on _Peek to Definition_ panel. How can i exclude them from _Definitions_ panel?

![Screen Shot 2023-09-28 at 20 34 37](https://github.com/DEVSENSE/phptools-docs/assets/2780617/dd419d7c-cda8-4d34-b9b5-d9fbd274dbce)

**Comment by heychazza** [comment]
> thank you for trying the `files.exclude` setting so quick; it looks like a bug, let me try it and we'll see what we can do

Hey, bump! Just bought the premium version and it's quite annoying that it shows vendor stubs as a "definition".

**Comment by jakubmisek** [comment]
@heychazza thank you for reminding us, we'll try to get back to it soon.

**Comment by jakubmisek** [comment]
This should be fixed in the latest release 1.54.

**Comment by jakubmisek** [comment]
I see this issue has been left unintended for a long time. As stated in https://github.com/DEVSENSE/phptools-docs/issues/780#issuecomment-2781399853 :

The next update will make this behavior consistent with VSCode

- `"files.exclude"` used to ignore files from indexing (and search, peek, references, workspace symbols, ...).
- `"search.exclude"` will be newly supported - makes the files ignored within _peek_, _find references_, _workspace symbols_.

**Comment by heychazza** [comment]
> I see this issue has been left unintended for a long time. As stated in [#780 (comment)](https://github.com/DEVSENSE/phptools-docs/issues/780#issuecomment-2781399853) :
> 
> The next update will make this behavior consistent with VSCode
> 
> * `"files.exclude"` used to ignore files from indexing (and search, peek, references, workspace symbols, ...).
> * `"search.exclude"` will be newly supported - makes the files ignored within _peek_, _find references_, _workspace symbols_.

Thanks a ton! Do files.exclude is the biggest one, whereas the second is indexed but excluded from search?

**Comment by jakubmisek** [comment]
@heychazza yes - "search.exclude" specifies file patterns that won't appear in searches (peek, workspace symbols, references, ...) (unless there is nothing else to show).
